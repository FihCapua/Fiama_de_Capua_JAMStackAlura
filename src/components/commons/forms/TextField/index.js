import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const InputWrapper = styled.div`
    display: flex;
    justify-content: center;
    margin: 17px 0; 
`;

const TextAreaWrapper = styled.div`
    display: flex;
    justify-content: center;
    margin: 17px 0; 
`;

const Input = styled.input`
  width: 78%;
  padding: 12px 16px;
  outline: 0;
  border: 1px solid #4d4d4d;
  border-radius: ${({ theme }) => theme.borderRadius};
`;

const TextAreaMsg = styled.textarea`
  width: 78%;
  height: 75px;
  padding: 12px 16px;
  outline: 0;
  border-radius: ${({ theme }) => theme.borderRadius};
  margin-top: 2px;
`;

Input.defaultProps = {
  tag: 'input',
  variant: 'paragraph1',
};

TextAreaMsg.defaultProps = {
  tag: 'textarea',
  variant: 'paragraph1',
};

export default function TextField({
  placeholder,
  name,
  onChange,
  value,
}) {
  return (
  // eslint-disable-next-line react/jsx-filename-extension
    <InputWrapper>
      <Input
        type="input"
        placeholder={placeholder}
        name={name}
        onChange={onChange}
        value={value}
      />
    </InputWrapper>
  );
}

export function TextArea({
  placeholder,
  name,
  onChange,
  value,
}) {
  return (
  // eslint-disable-next-line react/jsx-filename-extension
    <TextAreaWrapper>
      <TextAreaMsg
        type="textarea"
        placeholder={placeholder}
        name={name}
        onChange={onChange}
        value={value}
      />
    </TextAreaWrapper>
  );
}

TextField.propTypes = {
  placeholder: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
};

TextArea.propTypes = {
  placeholder: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
};
