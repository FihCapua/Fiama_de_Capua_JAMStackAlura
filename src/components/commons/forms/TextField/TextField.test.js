import React from 'react';
import { render, screen } from '../../../../infra/tests/testUtils';
import TextField from './index';

describe('<TextField />', () => {
  test('renders component', () => {
    render(
      <TextField
        placeholder="Nome"
        value="Fiama"
        onChange={() => {}}
        name="nome"
      />
      ,
    );

    const textField = screen.getByPlaceholderText(/nome/i);

    expect(textField).toMatchSnapshot();
  });
});
