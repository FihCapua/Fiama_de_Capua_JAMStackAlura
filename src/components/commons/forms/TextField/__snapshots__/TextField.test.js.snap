// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> renders component 1`] = `
.c0 {
  width: 78%;
  padding: 12px 16px;
  outline: 0;
  border: 1px solid #4d4d4d;
  border-radius: 5px;
}

<input
  class="c0"
  name="nome"
  placeholder="Nome"
  type="input"
  value="Fiama"
/>
`;
